### PR TITLE
[simdutf,scnlib] update to 4.0.9,2.0.2

### PR DESCRIPTION
**Pull request by jiayuehua**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by FrankXie05 — APPROVED** [review]

**Comment by data-queue** [comment]
What do these patches do? Can you submit them upstream?

**Comment by jiayuehua** [comment]
https://github.com/eliaskosunen/scnlib/pull/99 https://github.com/eliaskosunen/scnlib/pull/98 I have open two PR on upstream repo. one patch is fix SCN_HAS_STD_REGEX_MULTILINE  , only gcc 11.4 or above and clang 15.0.0 or above and support for std::regex_constants::multiline. The other is fix build error :

```
/mnt/vcpkg-ci/b/scnlib/src/v2.0.0-cc5e6bd45f.clean/src/scn/impl/reader/float_reader.cpp:275:22: error: no member named 'wcstod_l' in the global namespace; did you mean 'wcstold_l'?
            return ::wcstod_l(str, str_end, cloc);
```

and from /android-ndk-r25c/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/wchar.h we can see those functions are only supported on ANDROID_API >= 28

131 │ #if ANDROID_API >= 28
132 │ double wcstod_l(const wchar_t* __s, wchar_t** __end_ptr, locale_t __l) __INTRODUCED_IN(28);
133 │ #endif /* ANDROID_API >= 28 */

141 │ #if ANDROID_API >= 28
142 │ float wcstof_l(const wchar_t* __s, wchar_t** __end_ptr, locale_t __l) __INTRODUCED_IN(28);
143 │ #endif /* ANDROID_API >= 28 */

**Comment by eliaskosunen** [comment]
Patches for [scnlib](https://github.com/eliaskosunen/scnlib) merged upstream.

File: ports/scnlib/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
 {
   "name": "scnlib",
-  "version": "1.1.2",
+  "version": "2.0.0",
```
Comment (FrankXie05) [inline ports/scnlib/vcpkg.json:3]: The patch has been merged by the upstream, please update to the latest version and delete the patch.

https://github.com/eliaskosunen/scnlib/commit/a03ad30213cf13e4d18d18ca7534a4d3f07640ca

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". 
          That way, I can be aware that you've responded since you can't modify the tags.

File: ports/scnlib/vcpkg.json (hunk 1 above)
Comment (jiayuehua) [reply ports/scnlib/vcpkg.json]: I have update to 2.0.2, but still compile fail on gcc 11.3 due to macro SCN_HAS_STD_REGEX_MULTILINE, so I still add one patch. @eliaskosunen  @FrankXie05

**Comment by jiayuehua** [comment]
@FrankXie05 Please review this PR again.

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**